src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -412,6 +412,111 @@ public static DynamoModel Start(IStartConfiguration configuration)
             return new DynamoModel(configuration);
         }
 
+        private static FileVersion FindPreviousVersion(string rootFolder, FileVersionInfo currentVersionInfo)
```

Comment:
Could you explain the point of doing this? Are you taking into account the fact that a version path may exist but be empty, in which case, we have to continue searching for packages, etc. in all previous version paths until we find non -empty directories?